element-connector/src/main/java/org/eclipse/californium/elements/CorrelationContextMatcher.java

```diff
@@ -39,4 +42,15 @@
 	 */
 	boolean isResponseRelatedToRequest(CorrelationContext requestContext, CorrelationContext responseContext);
 
+	/**
+	 * Check, if message should be sent out using the current correlation
+	 * context of the connector.
+	 * 
+	 * @param messageContext correlation context of message
+	 * @param connectorContext correlation context of connector
+	 * @return true, if message should be sent, false, if message should not be
+	 *         sent.
+	 */
+	boolean isToBeSent(CorrelationContext messageContext, CorrelationContext connectorContext);
```

Comment:
is there really a difference in matching inbound vs. outbound messages?
At least `KeySetCorrelationContextMatcher` doesn't seem to bother whether it's inbound or outbound ...